src/rest/app.py

```diff
@@ -0,0 +1,178 @@
+    """Get all transcriptions and their statuses."""
+    transcriptions = []
+    for file_name in DATA_HANDLER.get_all_status_filenames():
+        try:
+            file_name = file_name[:-5]  # remove .json
+            data = DATA_HANDLER.get_status_file_by_id(file_name)
+            transcriptions.append(
+                {
+                    "transcription_id": data["transcription_id"],
+                    "status": data["status"],
+                }
+            )
+        except Exception as e:
+            LOGGER.error(f"Error while reading status file {file_name}: {e}")
+            DATA_HANDLER.delete_status_file(file_name)
+    return JSONResponse(content=transcriptions, status_code=200)
+
+
+@app.get("/transcriptions/{transcription_id}", dependencies=[Depends(require_api_key)])
+async def get_transcriptions_id(transcription_id: str):
+    """Get the status of a transcription by ID."""
+    file = DATA_HANDLER.get_status_file_by_id(transcription_id)
+    if file:
+        return JSONResponse(content=file, status_code=200)
+    raise HTTPException(status_code=404, detail="Transcription ID not found")
+
+
+@app.post("/transcriptions", dependencies=[Depends(require_api_key)])
+async def post_transcription(
+    file: UploadFile = File(...),
+    language: str | None = Form(None),
+    settings: str | None = Form(None),
+    model: str | None = Form(None),
+    task: str = Form("transcribe"),
+    text: str | None = Form(None),
+):
+    """Transcribe an audio file."""
+    if language and language not in config["supported_language_codes"]:
+        raise HTTPException(
+            status_code=400,
+            detail=f"Unsupported language code: {language}",
+        )
+
+    transcription_id = str(uuid.uuid4())
+    try:
+        audio = AudioSegment.from_file(file.file)
+        result = DATA_HANDLER.save_audio_file(audio, transcription_id)
+        if not result["success"]:
+            raise HTTPException(status_code=400, detail=result["message"])
+    except Exception as e:
+        LOGGER.error(f"Audio processing error: {e}")
+        raise HTTPException(status_code=500, detail="Something went wrong")
+
+    time.sleep(0.1)  # ensure file is saved
+
+    if task == "align" and (not text or not language):
+        missing_field = "text" if not text else "language"
+        raise HTTPException(
+            status_code=400,
+            detail=f"property {missing_field} is required for task align"
+        )
+
+    settings_dict = json.loads(settings) if settings else None
+    data = TranscriptionData(
+        transcription_id=transcription_id,
+        status=TranscriptionStatus.IN_QUERY.value,
+        start_time=datetime.utcnow().isoformat() + 'Z',
+        settings=settings_dict,
+        model=model,
+        task=task,
+        text=text,
+        language=language,
+    ).model_dump()
+
+    DATA_HANDLER.write_status_file(transcription_id, data)
+    return JSONResponse(content=data, status_code=200)
```

Comment:
How long does this request take? This seems like it may be very slow

Not really relevant for this PR though

Reply:
in other files are time_it decorators, we can go ahead and implement them everywhere to track statistics of the service itself if you would like?

Reply:
Lets leave it for now. Maybe in the long run we can make the decorator env dependant. Meaning we can toggle it on and off. Otherwise "instrumenting" every function is gonna bloat our logs. Also python may have a build in way to track function execution times on that scale